src/System.Net.Http/src/System/Net/Http/Unix/CurlResponseParseUtils.cs

```diff
@@ -73,29 +73,21 @@ internal static void ReadStatusLine(HttpResponseMessage response, string respons
             CheckResponseMsgFormat(index < responseHeader.Length && responseHeader[index] >= '0' && responseHeader[index] <= '9');
             int minorVersion = ReadInt(responseHeader, ref index);
 
-            SkipMandatorySpace(responseHeader, ref index);
+            SkipSpace(responseHeader, ref index, optional: false);
 
             //  parse status code
             int statusCode = ReadInt(responseHeader, ref index);
             CheckResponseMsgFormat(statusCode >= 100 && statusCode < 600);
 
-            SkipMandatorySpace(responseHeader, ref index);
+            SkipSpace(responseHeader, ref index, optional: true);
+            CheckResponseMsgFormat(index <= responseHeader.Length);
 
-            // we need reason phrase
-            CheckResponseMsgFormat( index < responseHeader.Length);
-
-            // Check if version is  HTTP/1.1 or HTTP/1.0
-            if (majorVersion == 1)
-            {
-                response.Version =
-                    minorVersion == 1 ? HttpVersion.Version11 :
-                    minorVersion == 0 ? HttpVersion.Version10 :
-                    new Version(0, 0);
-            }
-            else
-            {
-                response.Version = new Version(0, 0);
-            }
+            // Set the response HttpVersion
+            response.Version =
+                (majorVersion == 1 && minorVersion == 1) ? HttpVersion.Version11 :
+                (majorVersion == 1 && minorVersion == 0) ? HttpVersion.Version10 :
+                (majorVersion == 2 && minorVersion == 0) ? HttpVersion.Version20 :
+                new Version(0, 0);
```

Comment:
I realize this is just a simplification of what was there before, but should this be [`HttpVersion.Unknown`](https://github.com/dotnet/corefx/blob/2c6a6dc47ce711ced3705d09641a816d4903baa5/src/Common/src/System/Net/HttpVersion.cs#L10) now that `Unknown` is available on `HttpVersion`?